kotlin-options/src/main/kotlin/com/petertackage/kotlinoptions/Option.kt

```diff
@@ -12,16 +12,18 @@ object None : Option<Nothing>() {
 
 sealed class Option<out T : Any> {
 
-    fun ifSome(action: (T) -> Unit) {
-        when (this) {
-            is Some -> action(value)
+    inline fun ifSome(action: (T) -> Unit): Option<T> {
+        if (this is Some) {
+            action(value)
         }
+        return this
     }
 
-    fun ifNone(action: () -> Unit) {
-        when (this) {
-            is None -> action()
+    inline fun ifNone(action: () -> Unit): Option<T> {
+        if (this is None) {
```

Comment:
I know it's only one dimension to the solution, but is there a more idiomatic way of doing this? Perhaps something more consistent with the rest of the class? 

This what I could come up with:

```Kotlin

    fun ifSome(action: (T) -> Unit): Option<T> {
        return when (this) {
            is Some -> apply { action(value) }
            is None -> this
        }
    }

    fun ifNone(action: () -> Unit): Option<T> {
        return when (this) {
            is Some -> this
            is None -> apply { action() }
        }
    }
```